hashednode.go

```diff
@@ -30,69 +30,48 @@ import (
 	"fmt"
 )
 
-type HashedNode struct {
-	commitment  []byte
-	cachedPoint *Point
-}
+type HashedNode struct{}
 
-func (*HashedNode) Insert([]byte, []byte, NodeResolverFn) error {
+func (HashedNode) Insert([]byte, []byte, NodeResolverFn) error {
 	return errInsertIntoHash
 }
 
-func (*HashedNode) Delete([]byte, NodeResolverFn) (bool, error) {
+func (HashedNode) Delete([]byte, NodeResolverFn) (bool, error) {
 	return false, errors.New("cant delete a hashed node in-place")
 }
 
-func (*HashedNode) Get([]byte, NodeResolverFn) ([]byte, error) {
+func (HashedNode) Get([]byte, NodeResolverFn) ([]byte, error) {
 	return nil, errors.New("can not read from a hash node")
 }
 
-func (n *HashedNode) Commit() *Point {
-	if n.commitment == nil {
-		panic("nil commitment")
-	}
-	if n.cachedPoint == nil {
-		n.cachedPoint = new(Point)
-		n.cachedPoint.SetBytesUncompressed(n.commitment, true)
-	}
-	return n.cachedPoint
+func (HashedNode) Commit() *Point {
+	panic("can not commit a hash node")
 }
```

Comment:
Here's a tricky situation I'm not comfortable with, but I was forced by the API signature.

There's no real way to return something meaningful and correct in this API. In the case of `Empty` we could at least return the identity element. Here that would be wrong.

The reality is that `HashedNode` is now very hand-wavy on being a `VerkleNode` in the current "shape and form" of `VerkleNode` interface asking for `Commit()`, `Commitment()` and `Hash()`.

In practice, these panics should never trigger since any node that is `HashedNode` will be resolved before calling these kinds of APIs; but in the theoretical world of what's possible, I see these panics as a big risk.

This might require some rethought of our `VerkleNode` interface (maybe splitting in two, etc). We could also return something else (that would be incorrect; like the identity element) just to avoid a panic. But quite honestly, if a call to these methods happens that's 100% a bug.

Reply:
Added.